### PR TITLE
[Bug #3124] - clipped dropdown menu in Contents view

### PR DESCRIPTION
**Pull request by mihaislobozeanu**

### CODE REVIEW

**Review by avoinea — CHANGES_REQUESTED** [review]

File: theme/themes/pastanaga/modules/popup.overrides [hunk 1]
```diff
@@ -7,3 +7,63 @@
     vertical-align: middle;
   }
 }
+
+
+.dropdown-popup-trigger{
```
Comment (tiberiuichim) [inline theme/themes/pastanaga/modules/popup.overrides:12]: I think this file needs prettier for formatting. Usually there's a space before curlies.

**Review by avoinea — APPROVED** [review]

File: src/helpers/index.js [hunk 2]
```diff
@@ -94,3 +94,4 @@ export { useDetectClickOutside } from './Utils/useDetectClickOutside';
 export { usePrevious } from './Utils/usePrevious';
 export { usePagination } from './Utils/usePagination';
 export useUndoManager from './UndoManager/useUndoManager';
+export Popup from './Popup/Popup';
```
Comment (pnicolli) [inline src/helpers/index.js:97]: I thought about it after the team meeting today and I'm not a fan of having an actual component in the helpers folder. I would personally prefer moving it to the components folder. Just my 2 cents on this.

File: theme/themes/pastanaga/extras/contents.less [hunk 3]
```diff
@@ -80,6 +80,7 @@
 
   .contents-table-wrapper {
     width: 100%;
+    //#3124 - dropdown menus are clipped when table has few or no records
```
[the remark this answers is not in the document]
Comment (avoinea) [reply theme/themes/pastanaga/extras/contents.less:83]: I think this is a leftover. @mihaislobozeanu

**Comment by tiberiuichim** [comment]
@pnicolli We already have some components in helpers. https://github.com/plone/volto/blob/master/src/helpers/ScrollToTop/ScrollToTop.jsx and https://github.com/plone/volto/blob/master/src/helpers/BodyClass/BodyClass.jsx

I'm ok with moving it in another place. Maybe components/theme/PopupWrapper ?

**Comment by sneridagh** [comment]
I think also that it's not a helper, but a component... I would re-place it, what do you think? 

Also, but this is another question, what when we get rid of SemanticUI (eg. projects that do not use it at all? :)

**Comment by tiberiuichim** [comment]
> Also, but this is another question, what when we get rid of SemanticUI (eg. projects that do not use it at all? :)

when we get rid of semantic-ui we'll have other, bigger, things to worry about :-)

In any case, we'd begin with a refactoring of the Contents component and move it to react-table or something like that. I think that's something we should already start doing.

**Comment by avoinea** [comment]
@pnicolli @tiberiuichim What we agreed on this one?

**Comment by sneridagh** [comment]
If everybody agrees, we refactor it and move it to the components folder, then it's good to go?

**Comment by pnicolli** [comment]
> If everybody agrees, we refactor it and move it to the components folder, then it's good to go?

Agreed

**Comment by tiberiuichim** [comment]
I still don't think components/theme is the right place to put it, as it's a helper component that can be used even outside Volto. I can agree that src/helpers is not the best location, but there's no other existing place for it. Maybe a new `src/components/ui`. But I'm fine with placing it in src/components/theme, so +1 from me.

**Comment by sneridagh** [comment]
@tiberiuichim I agree with you that it's a new kind of component, and that ideally should be "separated" from the structural ones. I also thought in having this new category in the components folder for Quanta. See

https://github.com/plone/volto/pull/3086/files#diff-9837e4844c73b95722bd37d81ab4ce85e00a07552905224cfd7d9041e62703daR1

I initially called that "core", but naming... again. We can start doing it that now, or reorder them later.

**Comment by tiberiuichim** [comment]
In any case, we expose components through the `@plone/volto/components` index, so if we decide that we need to move it later, we can do that without too much breakage. Let's put it in components/theme.

**Comment by sneridagh** [comment]
@mihaislobozeanu could you please take care of the decision? Thanks!

**Comment by mihaislobozeanu** [comment]
> 

Resolved

**Review by giuliaghisini — APPROVED** [review]

File: src/components/manage/Contents/Contents.jsx [hunk 4]
```diff
@@ -618,7 +619,7 @@ class Contents extends Component {
    * @returns {undefined}
    */
   onChangeSelected(event, { value }) {
-    event.stopPropagation();
+    //event.stopPropagation();
```
Comment (sneridagh) [inline src/components/manage/Contents/Contents.jsx:622]: Are we sure about this one? and if so, let's remove it.

**Review by sneridagh — COMMENTED** [review]
@mihaislobozeanu LGTM, sorry for the pickiness, but could you take a look at the last thing? then let's merge.

**Pull request merged**